### ISSUE TITLE
Median completely incorrect

### ISSUE THREAD

**Issue by HalfVoxel** [liborty/medians#5]
I don't know if I'm misunderstanding something here, but this sure doesn't seem like a median to me? Note that the list is even sorted.

```rust
let a = vec![
    163.6170150950381,
    224.6127142531872,
    239.91368100304916,
    345.1674002412572,
    402.88833594261706,
    423.6406741377381,
    472.6292699764225,
    487.23306678749594,
    490.94434592125606,
    511.16658896980687,
    516.3472076946555,
    523.052566308903,
    563.6784311991111,
    586.7283185517608,
    633.5580942760708,
    678.4956618813414,
    708.2452516626092,
    741.9710552209048,
    763.476192474483,
    768.6249939324011,
    777.1952444919513,
    785.2192860329102,
    785.3178558989187,
    858.0319001781837,
    927.4228569429413,
    952.453888947949,
    1067.6089037099757,
];
eprintln!("Median: {} ", a.medf_checked().unwrap());
```
Prints:
```
Median: 163.6170150950381
```

Is there something I'm doing incorrectly?

Using the latest version 3.0.10

**Comment by liborty** [comment]
Looks like a bug, sorry, I will investigate and fix it.

**Comment by liborty** [comment]
I have fixed it now in the latest version 3.0.11
